kessel/auth/auth.go

```diff
@@ -0,0 +1,150 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/zitadel/oidc/v3/pkg/client"
+)
+
+const expirationWindow = 300  // 5 minutes in second
+const defaultExpiresIn = 3600 // 1 hour in seconds
+
+type OIDCDiscoveryMetadata struct {
+	TokenEndpoint string
+}
+
+type RefreshTokenResponse struct {
+	AccessToken string
+	ExpiresAt   time.Time
+}
+
+type OAuth2ClientCredentials struct {
+	clientId      string
+	clientSecret  string
+	tokenEndpoint string
+	cachedToken   RefreshTokenResponse
+	tokenMutex    sync.Mutex
+}
+
+type FetchOIDCDiscoveryOptions struct {
+	Context    context.Context
+	HttpClient *http.Client
+}
+
+type GetTokenOptions struct {
```

Comment:
Same comment/question about this as well as the DiscoveryOptions, it is not defined in the spec.

Reply:
I will add documentation inline, i don't think we should update the spec as this is specific to go. Only `ForceRefresh` is part of the spec.